fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/RecursiveUnionExpression.java

```diff
@@ -77,7 +78,19 @@ public int getRelationalChildCount() {
 
     @Nonnull
     @Override
-    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+    public Set<CorrelationIdentifier> computeCorrelatedTo() {
```

Comment:
I take it that this was a bug that was in the `RecursiveUnionExpression` that meant is was miscalculating its correlated to sets? Did we add tests that surface this problem?

Reply:
done.